### PR TITLE
Update concourse template to deploy concourse v3.12.0

### PR DESCRIPTION
**Pull request by norshtein**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog
- Support deploy concourse automatically
- Upgrade versions
    - Upgrade Bosh version to 265.2.0, with manifest files changed
    - Upgrade Concourse version to 3.12.0, with manifest files changed
    - Upgrade Azure CPI version to 35.0.0, with manifest files changed

Hi, it is noticed the latest Concourse version is v3.14.1, but the concourse-ci template still deploys Concourse v2.6.0, which is out of date. I have upgraded the template to deploy Concourse v3.12.0 and manually validated the template. The new version Concourse should be more stable and easier to use.

### CODE REVIEW

File: concourse-ci/azuredeploy.json [hunk 1]
```diff
@@ -73,6 +73,27 @@
         "description": "secret of the client. See https://github.com/cloudfoundry-incubator/bosh-azure-cpi-release/blob/master/docs/guidance.md"
       }
     },
+    "boshVmSize": {
+      "type": "string",
+      "defaultValue": "Standard_D2_v2",
+      "allowedValues": [
```
Comment (bmoore-msft) [inline concourse-ci/azuredeploy.json]: remove allowedValues

File: concourse-ci/azuredeploy.parameters.json [hunk 2]
```diff
@@ -32,8 +32,14 @@
     "clientSecret": {
       "value": "GEN-PASSWORD"
     },
+    "boshVmSize": {
```
Comment (bmoore-msft) [inline concourse-ci/azuredeploy.parameters.json]: remove this and just use a defaultValue in the template

File: concourse-ci/azuredeploy.parameters.json [hunk 3]
```diff
@@ -32,8 +32,14 @@
...
+      "value": "Standard_D2_v2"
+    },
     "autoDeployBosh": {
-      "value": "disabled"
+      "value": "enabled"
+    },
+    "autoDeployConcourse": {
```
Comment (bmoore-msft) [inline concourse-ci/azuredeploy.parameters.json]: remove this and just use a defaultValue in the template

File: concourse-ci/azuredeploy.parameters.json [hunk 4]
```diff
@@ -32,8 +32,14 @@
     "clientSecret": {
       "value": "GEN-PASSWORD"
     },
+    "boshVmSize": {
+      "value": "Standard_D2_v2"
+    },
     "autoDeployBosh": {
-      "value": "disabled"
+      "value": "enabled"
```
Comment (bmoore-msft) [inline concourse-ci/azuredeploy.parameters.json]: remove this and just use a defaultValue in the template

File: concourse-ci/metadata.json [hunk 5]
```diff
@@ -3,7 +3,7 @@
   "itemDisplayName": "Concourse CI",
   "description": "Concourse is a CI system composed of simple tools and ideas. It can express entire pipelines, integrating with arbitrary resources, or it can be used to execute one-off tasks, either locally or in another CI system. This template can help to prepare neccessary Azure resources to setup such a CI system, and make the setup more simple.",
   "summary": "This template helps you setup Bosh and Concourse CI.",
-  "githubUsername": "bingosummer",
-  "dateUpdated": "2016-08-11"
+  "githubUsername": "norshtein",
```
Comment (bmoore-msft) [inline concourse-ci/metadata.json]: are you taking ownership of this sample?

Comment (bingosummer) [reply concourse-ci/metadata.json]: yes. @norshtein will take the ownership of this template.

**Pull request merged**